### ISSUE TITLE
How does the instruction cache of Snitch work?

### ISSUE THREAD

**Issue by rgantonio** [KULeuven-MICAS/snitch_cluster#28]
We bring up this issue because there are uncontrollable stalls that were seen in hardware.

One concrete example is when we run the GEMM tests where one is running the base MM operation and one is running the GEMM operation.

In each test there is a DMA loading sequence that is the same in both tests; however, both end up in different cycle numbers. Initial investigation shows that the instruction cache is stalling the GEMM version. Difficult to trace, but our hunch is that we ended up in an instruction miss and hence stalls the core for a few cycles just to fetch the correct cycle.

Basically, we need to just understand how the Snitch instruction cache works!
